tests/integration/components/file-input-test.js

```diff
@@ -0,0 +1,139 @@
+import { render, find }       from '@ember/test-helpers';
+
+import hbs                    from 'htmlbars-inline-precompile';
+import { setupRenderingTest } from 'ember-qunit';
+import { module, test }       from 'qunit';
+
+module('Integration | Component | {{file-input}}', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function(assert) {
+    await render(hbs`{{file-input}}`);
+    assert.ok(find('button'), 'shows a button');
+    assert.ok(find('button > i.fa.fa-upload'), 'shows default icon in button');
+    assert.ok(find('input[type=file]'), 'Has (hidden) file input');
+    assert.equal(this.element.textContent.trim(), 'No file selected',
+      'shows default message');
+  });
+
+  test('sets `accept` attribute on input', async function(assert) {
+    await render(hbs`{{file-input}}`);
+
+    let $input = find('input');
+    assert.equal($input.getAttribute('accept'), '*/*',
+      'accept all files by default');
+
+    this.set('type', 'foobar');
+    await render(hbs`{{file-input type=type}}`);
+    $input = find('input');
+
+    assert.equal($input.getAttribute('accept'), '*/*',
+      'accept all files by when type is not a regular file type');
+
+    this.set('type', 'image');
+    assert.equal($input.getAttribute('accept'), 'image/*',
+      'accept all image files by when type is image');
+
+    this.set('type', 'video');
+    assert.equal($input.getAttribute('accept'), 'video/*',
+      'accept all video files by when type is video');
+
+    this.set('type', 'audio');
+    assert.equal($input.getAttribute('accept'), 'audio/*',
+      'accept all audio files by when type is audio');
+
+    this.set('accept', '.foo');
+    await render(hbs`{{file-input accept=accept}}`);
+    $input = find('input');
+    assert.equal($input.getAttribute('accept'), '.foo',
```

Comment:
I think this should be a separate test: 'Can override default accept', something like that.